editor/editor_properties_array_dict.cpp

```diff
@@ -111,6 +111,7 @@ bool EditorPropertyDictionaryObject::_set(const StringName &p_name, const Varian
 	}
 
 	if (name.begins_with("indices")) {
+		dict = dict.duplicate();
```

Comment:
Why is this needed?

Reply:
This is needed for undo/redo so that the new and old dictionaries are different and uptading the new won't update the old. It didn't cause any bug before because in fact it was never called. I realized there was duplication of code between this function and one part of code around line 750. The only difference being it never duplicated the dict in the function. So I added the duplication and removed the line of codes to call set instead.